lib/iex/test/iex/autocomplete_test.exs

```diff
@@ -162,4 +162,33 @@ defmodule IEx.AutocompleteTest do
     assert expand('EList') == {:yes, '.', []}
     assert expand('EList.map') == {:yes, [], ['map/2', 'mapfoldl/3', 'mapfoldr/3']}
   end
+
+  test "completion for functions added when a module is reloaded" do
+    filename = "sample.ex"
+    content = """
+    defmodule Sample do
+    def foo(), do: 0
+    end
+    """
+
+    reloaded_content = """
+    defmodule Sample do
+    def foo(), do: 0
+    def foobar(), do: 0
+    end
+    """
+    File.write!(filename, content)
+    try do
+      :elixir_compiler.file_to_path(filename, ".")
+      assert expand('Sample.foo') == {:yes, '', ['foo/0']}
+      File.write!(filename, reloaded_content)
+      Code.load_file(filename)
```

Comment:
Nevermind, I managed to avoid the warning by temporarily changing the compiler options.